### ISSUE TITLE
Slowdowns noticed after panda install

### ISSUE THREAD

**Issue by dwarring** [pdf-raku/PDF-raku#15]
I noticed this when I was trying to profile code:

```
$ perl6 -v
This is Rakudo version 2016.03-105-ga3b8fef built on MoarVM version 2016.03-84-g4afd7b6
implementing Perl 6.c.
$ perl6 -I lib -e'use PDF::DAO;PDF::DAO.coerce: :stream{} for 1..10'
$ time perl6 -I lib -e'use PDF::DAO;PDF::DAO.coerce: :stream{} for 1..50'

real    0m2.875s
user    0m2.740s
sys     0m0.132s
```

But when I drop the `-I lib` from the command line, and use the same version, panda installed:

```
$ perl6 -e'use PDF::DAO;PDF::DAO.coerce: :stream{} for 1..10'
$ time perl6 -e'use PDF::DAO;PDF::DAO.coerce: :stream{} for 1..50'

real    0m25.763s
user    0m25.564s
sys     0m0.188s
```

Invocations of the coerce method are much slower. I suspect a rakudo precompilation issue, but more investigation is needed,

**Comment by dwarring** [comment]
I get similar results after reinstalling PDF-Tools with zef

**Comment by dwarring** [comment]
Slows down just as much with a simple int coercement:
`time perl6 -I lib -e'use PDF::DAO; PDF::DAO.coerce: :int(42) for 1..50'`

**Comment by dwarring** [comment]
Golfed it down, it's the dynamic `require ::($foo)` that's run slow:

```
$ perl6 -I lib -e'require ::("PDF::DAO::Stream")'; time perl6 -I lib -e'for 1..20 { require ::("PDF::DAO::Stream") }'

real    0m1.674s
user    0m1.552s
sys     0m0.120s
```

Versus:

```
$ perl6 -e'require ::("PDF::DAO::Stream")'; time perl6 -e'for 1..20 { require ::("PDF::DAO::Stream") }'
real    0m10.952s
user    0m10.752s
sys     0m0.200s
```

I can both work-around and rakudobug this.

**Comment by dwarring** [comment]
Have raised rakudobug https://rt.perl.org/Ticket/Display.html?id=127873
Also added method DAO.required("My", "Module") with local caching to mitigate this.

``````
$ perl6 -M PDF::DAO -M Bench -e'PDF::DAO.required("PDF::DAO::Stream"); Bench.new.timethese(50, {work-around => sub { PDF::DAO.required("PDF::DAO::Stream") }, bottleneck => { require ::("PDF::DAO::Stream")}})'
Benchmark: 
Timing 50 iterations of bottleneck, work-around...
bottleneck: 22.4367 wallclock secs @ 2.2285/s (n=50)
work-around: 0.0091 wallclock secs @ 5510.0957/s (n=50)
                (warning: too few iterations for a reliable count)```
``````

**Comment by dwarring** [comment]
Regression has gone in latest rakudo. Closed along with rt#127873
